rust/agama-lib/share/profile.schema.json

```diff
@@ -251,22 +251,156 @@
+              "type": "array",
+              "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "required": ["mount"],
+                "properties": {
+                  "mount": {
+                    "type": "object",
+                    "additionalProperties": false,
+                    "required": ["path"],
+                    "properties": {
+                      "path": {
+                      "title": "Mount path",
+                      "type": "string"
+                      },
+                      "options": {
+                        "title": "Options to add to the fourth field of fstab",
+                        "type": "string"
+                      }
+                    }
+                  },
+                  "filesystem": {
+                    "title": "File system of the volume",
+                    "type": ["string", "object"],
+                    "additionalProperties": false,
+                    "properties": {
+                      "btrfs": {
+                        "title": "Specification of a Btrfs file system",
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                          "snapshots": {
+                            "title": "Whether Btrfs snapshots should be configured",
+                            "type": "boolean"
+                          }
+                        }
+                      }
+                    }
+                  },
+                  "size": {
+                    "title": "Size limits",
+                    "description": "It can be 'auto', a fixed size (as string or integer) or a set of size limits",
+                    "type": ["string", "integer", "array", "object"],
+                    "prefixItems": [
+                      {
+                        "title": "Mandatory lower size limit",
+                        "type": ["integer", "string"]
+                      },
+                      { 
+                        "title": "Optional upper size limit",
+                        "type": ["integer", "string"]
+                      }
+                    ],
+                    "additionalProperties": false,
+                    "properties": {
+                      "min": {
+                        "title": "Mandatory lower size limit",
+                        "type": ["integer", "string"]
+                      },
+                      "max": {
+                        "title": "Optional upper size limit",
+                        "type": ["integer", "string"]
+                      }
+                    }
+                  },
+                  "target": {
+                    "title": "Location of the resulting file system",
+                    "description": "Can be 'default' or an object with exactly one key-value pair",
+                    "type": ["string", "object"],
```

Comment:
This is another candidate for `oneOf`.

Reply:
Done in the amended commit (as said, I force-pushed because I overlooked your comments).